src/util/system.cpp

```diff
@@ -578,75 +577,82 @@ fs::path GetDefaultDataDir()
     else
         pathRet = fs::path(pszHome);
 #ifdef MAC_OSX
-    // Mac
+    // macOS
     return pathRet / "Library/Application Support/Bitcoin";
 #else
-    // Unix
+    // Unix-like
     return pathRet / ".bitcoin";
 #endif
 #endif
 }
 
-static fs::path g_blocks_path_cache_net_specific;
-static fs::path pathCached;
-static fs::path pathCachedNetSpecific;
-static RecursiveMutex csPathCached;
+static Mutex g_path_cache_mutex;
+static fs::path g_blocks_path_cache_net_specific GUARDED_BY(g_path_cache_mutex);
+static fs::path pathCached GUARDED_BY(g_path_cache_mutex);
+static fs::path pathCachedNetSpecific GUARDED_BY(g_path_cache_mutex);
 
-const fs::path &GetBlocksDir()
+namespace {
```

Comment:
Why is this `namespace` required?

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Rs-unnamed2